provider/spark.go

```diff
@@ -701,8 +708,54 @@ func (spark *SparkOfflineStore) GetResourceTable(id ResourceID) (OfflineTable, e
 	}}, nil
 }
 
+type S3Materialization struct {
+}
+
+func (s *S3Materialization) ID() MaterializationID {
+	return nil
+}
+
+func (s *S3Materialization) NumRows() (int64, error) {
+	return 0, nil
+}
+
+func (s *S3Materialization) IterateSegment(begin, end int64) (FeatureIterator, error) {
+	return &S3FeatureIterator{}, nil
+}
+
+type S3FeatureIterator struct {
+}
+
+func (s *S3FeatureIterator) Next() bool {
+	return false
+}
+
+func (s *S3FeatureIterator) Value() ResourceRecord {
+	return nil
+}
+
+func (s *S3FeatureIterator) Err() error {
+	return nil
+}
+
 func (spark *SparkOfflineStore) CreateMaterialization(id ResourceID) (Materialization, error) {
-	return nil, nil
+	if _, err := spark.GetResourceTable(id); err != nil {
+		return nil, fmt.Errorf("Resource not registered: %v", err)
+	}
+	materializationID := ResourceID{Name: id.Name, Variant: id.Variant, Type: FeatureMaterialization}
```

Comment:
Whats the difference between Feature and FeatureMaterialization?

Reply:
featurematerialization is just a materialization, i just named it that since Materialization is already a struct type in offline.go

Reply:
Why don't the other offline stores use this?

Reply:
they store materializations under the separate category materialized view. For S3 paths to distinguish materializations we need a new type string.

Reply:
Whats the separate category? Aren't they all just tables?

Reply:
yes, I was looking at the postgres implementation and there it's a materialized view rather than a table. The way I do paths for each resource is {type}/{name}/{variant}. Any materialization of a feature would have the same name and variant so I created a new type so they are stored in seperate pseudo S3 directories. It felt low-risk to use the existing OfflineResourceType convention as a path prefix for each type stored in S3.

Reply:
Got it